### DISCUSSION TITLE
Thinner UI

### DISCUSSION THREAD

**Discussion by matthewmueller** [zed-industries/zed#23933] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

This is fairly minor, but I'd love to be able to minimize the chrome to maximize the working space of the editor. Zed has a really clean design, but I still feel like there's quite a bit of extra space that would be nice to collapse, specifically:
- Despite disabling code folding, the column takes up space. Ideally when you disable column features, the space is reclaimed.
- The scrollbars don't match. I really like how thin the scrollbar is on the tree view and would love to have that same treatment on the code view
- I'd love to be able to hide the tabs up at the top. I typically navigate code with cmd+p and the back and forward keyboard shortcuts. Again to reclaim space.

### If applicable, add mockups / screenshots to help present your vision of the feature

![CleanShot 2024-10-22 at 00 26 53@2x](https://github.com/user-attachments/assets/762b32aa-54f7-470d-9d8c-056c83cab753)

This is my first time trying Zed, I'm amazed how polished the experience is already. Keep up the great work!

**Comment by bbb651** [comment]
Disabling folding, with:
```json
"gutter": {
    "folds": false
},
```
does significantly reduce the space in the gutter (to the right of the line numbers, as can be seen in your screenshot). You might want to disable `gutter.runnables` as well, it significantly reduces the width of the gutter as well (and personally I disable `gutter.code_actions` because I always use the shortcut and I find it annoying, although it doesn't take additional space).

The non-floating [editor scrollbar](https://zed.dev/docs/configuring-zed#editor-scrollbar) is on purpose, it's used to display information about the code (git status, diagnostics, symbols, etc.) and uses third columns to separate the different types. I don't think you can turn it floating but you can hide it completly:
```json
"scrollbar": {
    "show": "never"
}
``` 

You can hide tabs with [`tab_bar`](https://zed.dev/docs/configuring-zed#editor-tab-bar):
```json
"tab_bar": {
  "show": false,
}
```
(or just the back and forward buttons with `tab_bar.show_nav_history_buttons`)

**Comment by matthewmueller** [comment]
Thanks! I didn't know about disabling the tab bar, that helps. You're right that if you disable all left-hand gutter items it does reduce the left hand side. Would love for the numbers to hug the line a bit more, but that's being nit picky.

I still think there is quite a bit of room to reduce the size of the code scrollbar. For comparison, here's what VSCode looks like for me:
![CleanShot 2024-10-22 at 21 52 28@2x](https://github.com/user-attachments/assets/f834ea55-fc67-45cb-ba0b-2f7d65857f05)

**Comment by AndydeCleyre** [comment]
I'm not sure of the scope of discussion for this issue, but . . .

I wish I could completely dissolve the "toolbar" (breadcrumbs + quick actions) and "title bar" (hamburger, recent projects, branches, sign in, settings), without losing all the elements:

- The hamburger, sign in, recent projects, and settings menu should all be exported as a global menu in desktops which support that. Or hide-able, as I don't use any of them. 
- The branches and breadcrumbs should be optional elements of the status bar, which has tons of unused space, and those elements reflect current status.
- The quick actions buttons could easily join the tab bar or status bar -- I don't care, I'd hide them either way.

In the screenshot, the red marks items or spaces I don't ever use, and the blue marks the items suitable for the status bar, and the free space on the status bar itself.

![image](https://github.com/user-attachments/assets/5c287b0e-2515-430f-ab40-96b8ad7016a4)
